### PR TITLE
[THORN-2561] config-1.4, typos

### PR DESCRIPTION
**Pull request by penehyba**
Testing built-in converters for byte, short and char.

https://download.eclipse.org/microprofile/microprofile-config-1.4/microprofile-config-spec.html#release_notes_14

### CODE REVIEW

File: microprofile/microprofile-config-1.2/src/main/resources/META-INF/microprofile-config.properties [hunk 1]
```diff
@@ -1,3 +0,0 @@
-app.timeout=314159
-fileProperty=This will be overriden by a file
-yamlOrdereredProperty=Property from default config source
```
Comment (Ladicek) [inline microprofile/microprofile-config-1.2/src/main/resources/META-INF/microprofile-config.properties:3]: Why is this deleted?

File: microprofile/microprofile-config-1.3/src/main/resources/META-INF/microprofile-config.properties [hunk 2]
```diff
@@ -1,5 +1,2 @@
-app.timeout=314159
-fileProperty=This will be overriden by a file
-yamlOrdereredProperty=Property from default config source
```
Comment (Ladicek) [inline microprofile/microprofile-config-1.3/src/main/resources/META-INF/microprofile-config.properties:3]: Why is this deleted?

File: microprofile/microprofile-config-1.4/src/main/resources/project-defaults.yml [hunk 3]
```diff
@@ -0,0 +1,12 @@
+swarm:
+  microprofile:
+    config:
+      config-sources:
+        default:
+          dir: ${project.build.directory}/classes/configs
```
Comment (Ladicek) [inline microprofile/microprofile-config-1.4/src/main/resources/project-defaults.yml:6]: This config source isn't used, is it?

**Comment by Ladicek** [comment]
Otherwise LGTM.

File: microprofile/microprofile-config-1.4/src/main/resources/project-defaults.yml (hunk 3 above)
Comment (penehyba) [reply microprofile/microprofile-config-1.4/src/main/resources/project-defaults.yml:6]: right, it will be removed

File: microprofile/microprofile-config-1.2/src/main/resources/META-INF/microprofile-config.properties (hunk 1 above)
Comment (penehyba) [reply microprofile/microprofile-config-1.2/src/main/resources/META-INF/microprofile-config.properties:3]: unused... I fixed the typo and fixing unused typo seemed insane :)
Do you prefer empty microprofile-config.properties instead of removing it? (in config-1.2 tests)

File: microprofile/microprofile-config-1.3/src/main/resources/META-INF/microprofile-config.properties (hunk 2 above)
Comment (penehyba) [reply microprofile/microprofile-config-1.3/src/main/resources/META-INF/microprofile-config.properties:3]: 3 unused properties removed, 2 remaining are used

File: microprofile/microprofile-config-1.2/src/main/resources/META-INF/microprofile-config.properties (hunk 1 above)
Comment (Ladicek) [reply microprofile/microprofile-config-1.2/src/main/resources/META-INF/microprofile-config.properties:3]: No, I meant all 3 properties. Not sure about `app.timeout`, but the `fileProperty` is specifically there to be sure that it is overridden by something else.

Comment (penehyba) [reply microprofile/microprofile-config-1.2/src/main/resources/META-INF/microprofile-config.properties:3]: this is used in v11... in v12 it is unused... otherwise the test would fail and I would find it in fulltext in this v12 module.. all 3 are used in org.wildfly.swarm.ts.microprofile.config.v11.HelloServlet
and unused in 
org.wildfly.swarm.ts.microprofile.config.v12.HelloServlet
Trust me in this... anyway it would/will fail in tests :)

Comment (Ladicek) [reply microprofile/microprofile-config-1.2/src/main/resources/META-INF/microprofile-config.properties:3]: That is weird, I ran the entire TS once or twice, without this change, and it passed...

Comment (penehyba) [reply microprofile/microprofile-config-1.2/src/main/resources/META-INF/microprofile-config.properties:3]: I dont argue that this fixes something.. it is just unused and I ran into it when fixing typos.. no big deal

Comment (Ladicek) [reply microprofile/microprofile-config-1.2/src/main/resources/META-INF/microprofile-config.properties:3]: Well I am arguing that some of these config properties are present with the intention that they will be overridden somewhere else, and so if the test observed a `This will be overriden by a file`, it would be a natural failure.

Comment (Ladicek) [reply microprofile/microprofile-config-1.2/src/main/resources/META-INF/microprofile-config.properties:3]: But I'm not sure if all are like this.

Comment (penehyba) [reply microprofile/microprofile-config-1.2/src/main/resources/META-INF/microprofile-config.properties:3]: In v11 the "app.timeout" is used in servlet:
https://github.com/rhoar-qe/thorntail-test-suite/blob/123e94f997846fdcbf9c4103e2fc8665b1c19a7c/microprofile/microprofile-config-1.1/src/main/java/org/wildfly/swarm/ts/microprofile/config/v11/HelloServlet.java#L16
and verified in test:
https://github.com/rhoar-qe/thorntail-test-suite/blob/123e94f997846fdcbf9c4103e2fc8665b1c19a7c/microprofile/microprofile-config-1.1/src/test/java/org/wildfly/swarm/ts/microprofile/config/v11/MicroProfileConfig11Test.java#L22
loaded from microprofile-config.properties as expected and in v11 this stays the same (PR doesn't modify this).

In v12 the string "app.timeout" appears only in microprofile-config-1.2/src/main/resources/META-INF/microprofile-config.properties, this is not used by any annotation, parameter, not loaded and it is not tested. This is most likely just copy of microprofile-config-1.1/src/main/resources/META-INF/microprofile-config.properties from previous version (v11).

In v13 the same situation (microprofile-config-1.3/src/main/resources/META-INF/microprofile-config.properties).

This applies to all 3 properties.
This removal of "app.timeout","fileProperty","yamlOrdereredProperty" in v12 and v13 has no effect to tests because these properties were not used and affects nothing in v12 and v13.

Comment (Ladicek) [reply microprofile/microprofile-config-1.2/src/main/resources/META-INF/microprofile-config.properties:3]: Ah, ok! Thanks, didn't understand it on the first try :-)

**Comment by Ladicek** [comment]
Thanks!
